### PR TITLE
Add displayAttribute field to user schema model, DB, and store

### LINKED ISSUE #1512: Add `systemAttributes` support to user schema model, DB, and store
Parent issue: #1511

## Scope

Add the `systemAttributes` JSON object to user schemas across the model layer, database scripts, store queries, handler sanitization, and declarative resource loader. The `systemAttributes` object initially contains a `display` field and is stored as a JSON column for future extensibility.

## API Surface

```json
{
    "id": "019cb209-ce92-7142-aff9-d822a3979d8e",
    "name": "Person",
    "systemAttributes": {
        "display": "email"
    },
    "schema": {
        "email": { "type": "string", "required": true, "unique": true },
        "password": { "type": "string", "required": true, "credential": true }
    }
}
```

## Tasks

- [x] Add `SystemAttributes` struct with `Display string` field to model
- [x] Add `SystemAttributes *SystemAttributes` to model structs: `UserSchema`, `UserSchemaListItem`, `CreateUserSchemaRequest`, `UpdateUserSchemaRequest`, `UserSchemaRequestWithID`
- [x] Add `SYSTEM_ATTRIBUTES` column to `USER_SCHEMAS` table in `sqlite.sql` (`TEXT`, nullable) and `postgres.sql` (`JSONB`, nullable)
- [x] Update store SQL queries:
  - `queryGetUserSchemaList` — add `SYSTEM_ATTRIBUTES` to SELECT
  - `queryCreateUserSchema` — add `SYSTEM_ATTRIBUTES` as new parameter
  - `queryGetUserSchemaByID` — add `SYSTEM_ATTRIBUTES` to SELECT
  - `queryGetUserSchemaByName` — add `SYSTEM_ATTRIBUTES` to SELECT
  - `queryUpdateUserSchemaByID` — add `SYSTEM_ATTRIBUTES` SET clause
- [x] Update `parseUserSchemaFromRow` and `parseUserSchemaListItemFromRow` to parse `SYSTEM_ATTRIBUTES` (handle NULL, JSON unmarshal)
- [x] Update `CreateUserSchema` and `UpdateUserSchemaByID` store methods to JSON marshal and pass `SystemAttributes`
- [x] Add `SystemAttributes` sanitization in handler (`sanitizeCreateUserSchemaRequest`, `sanitizeUpdateUserSchemaRequest`) — sanitize `Display` field via `sysutils.SanitizeString()`
- [x] Map `SystemAttributes` in `parseToUserSchemaDTO` in declarative resource loader
- [x] Map `SystemAttributes` in file-based store if applicable

## Files

- `backend/internal/userschema/model.go`
- `backend/internal/userschema/store_constants.go`
- `backend/internal/userschema/store.go`
- `backend/internal/userschema/handler.go`
- `backend/internal/userschema/declarative_resource.go`
- `backend/internal/userschema/file_based_store.go`
- `backend/dbscripts/thunderdb/sqlite.sql`
- `backend/dbscripts/thunderdb/postgres.sql`

## Design Discussion

#1510

### PR DESCRIPTION
**Pull request by DonOmalVindula**
## Summary

- Add `DISPLAY_ATTRIBUTE VARCHAR(100)` nullable column to `USER_SCHEMAS` table (SQLite + PostgreSQL)
- Add `DisplayAttribute` field to all user schema model structs (`UserSchema`, `UserSchemaListItem`, `CreateUserSchemaRequest`, `UpdateUserSchemaRequest`, `UserSchemaRequestWithID`)
- Update all store SQL queries (SELECT, INSERT, UPDATE) to include the new column
- Update row parsers to handle the nullable column with `nullableString`/`parseNullableString` helpers
- Update handler sanitization to sanitize `DisplayAttribute` input
- Update declarative resource loader and file-based store to pass through the field

## Test plan

- [x] Verify Go build passes (`go build ./...`)
- [x] Verify existing userschema tests pass (`go test ./internal/userschema/...`)
- [x] Verify user schema CRUD operations work with and without `displayAttribute`
- [x] Verify existing schemas without `displayAttribute` continue to work (nullable column)

Resolves #1512
Parent issue: #1511

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * DisplayAttribute added to user schema API: settable on create/update, sanitized, persisted, and shown in schema listings.

* **Chores**
  * Database schemas updated so DisplayAttribute is stored across supported databases.

* **Tests**
  * Added unit tests covering parsing, listing, creation/update handling, and edge cases for DisplayAttribute and related parsing helpers.

### CODE REVIEW

File: backend/internal/userschema/service.go [hunk 1]
```diff
@@ -156,6 +156,7 @@ func (us *userSchemaService) CreateUserSchema(
 		Name:                  request.Name,
 		OrganizationUnitID:    request.OrganizationUnitID,
 		AllowSelfRegistration: request.AllowSelfRegistration,
+		DisplayAttribute:      request.DisplayAttribute,
```
Comment (thiva-k) [inline backend/internal/userschema/service.go]: Do we need to add any validations here? This should be a string attribute from the schema that is not a credential right?

Comment (DonOmalVindula) [reply backend/internal/userschema/service.go]: validations will be added in a follow-up PR - https://github.com/asgardeo/thunder/pull/1525

I have split the implementation into simpler chunks
